### DISCUSSION TITLE
Should new keys automatically appear in the tolgee platform?

### DISCUSSION THREAD

**Discussion by flyon** [tolgee/tolgee-platform#1999] [Q&A]
I followed the installation guides, and if I understand correctly the keys I add in my code should show up in the tolgee platform right?

I've created a project in the platform, then added several keys in the code and I see requests in the network tab, but they return an empty object.

I was expecting new keys would automatically be added but maybe that's not the case?

Am I missing something? 

Here's my setup.

utils/tolgee.ts
```tsx
import { Tolgee, DevTools, BackendFetch} from "@tolgee/react";
import { FormatIcu } from '@tolgee/format-icu';
export function initTolgee() {
  const tolgee = Tolgee()
    .use(DevTools())
    .use(BackendFetch())
    .use(FormatIcu())
    .init({
      language: 'en',

      // for development
      apiUrl: process.env.REACT_APP_TOLGEE_API_URL,
      apiKey: process.env.REACT_APP_TOLGEE_API_KEY,
    });
  return tolgee;
}
```
(the env variables are defined with the key and url provided by the tolgee platform)

App.tsx
```tsx
import { initTolgee } from './utils/tolgee';
import { TolgeeProvider } from '@tolgee/react';

let tolgee = initTolgee();

//....
<TolgeeProvider tolgee={tolgee}>
    <AppRoutes />
</TolgeeProvider />
```

Component.tsx
```tsx
import { useTranslate } from "@tolgee/react";
export const Foo = () => {
    let {t} = useTranslate();
 return <div>
  <h1>{t('testkey1')}</h1>
  <p>{t('testkey1')}</p>
</div>;
}
```

**Comment by flyon** [comment]
I realized now the keys only show up in the platform once I open the context menu (with Option) and add a translation / save the key. 

I think it would be nice to automatically save the key?

**Reply by stepan662** [reply]
Hey, we were thinking about that, but it's hard to determine when exactly the new key should be saved. As you can be in process of designing the page and you put a temporary placeholder or something, the we'd clutter the platform with useless keys.

What I'm thinking we might add some "add all" feature, where it would list all non-translated keys on the page and let you save them all at once. That might be a good approach for the future.

**Reply by flyon** [reply]
I agree, that would be good idea. Alternatively, yes the space can get cluttered when you auto add keys. But you could also automatically remove keys that are not present in the code and also do not have any translations? 

Either way, the best way for now is to manually open the context menu on each translated text in the app and press save, correct?

**Reply by stepan662** [reply]
Yes, now you need to do it manually
